hail/src/main/scala/is/hail/expr/ir/ExtractAggregators.scala

```diff
@@ -142,13 +142,32 @@ object ExtractAggregators {
                 ArrayRange(I32(0), ArrayLen(Ref(aUID, a.typ)), I32(1)),
                 iUID,
                 Let(
-                  name,
+                  elementName,
                   ArrayRef(Ref(aUID, a.typ), Ref(iUID, TInt32())),
-                  SeqOp(
+                  addLets(SeqOp(
                     I32(i),
                     FastIndexedSeq(Ref(iUID, TInt32()), Begin(seqOp.toFastIndexedSeq)),
-                    aggSig)))))), dependent))
-        ArrayMap(GetTupleElement(result, i), newRef.name, transformed)
+                    aggSig), dependent)
+                ))))))
+        val rUID = genUID()
+        Let(
+          rUID,
+          GetTupleElement(result, i),
+          ArrayMap(
+            ArrayRange(
+              I32(0),
+              ArrayLen(Ref(rUID, rt)),
+              I32(1)),
+            indexName,
+            Let(
+              newRef.name,
```

Comment:
Is this mapping over each aggregator and that's what `indexName` corresponds to? Index in the aggregator array?

Reply:
no, I had to add this to support the middle line here:

```
              .aggElements('__element_idx, '__result_idx)(
                  let(sa = 'global(colsField)('__result_idx)) {
                    aggLet(sa = 'global (colsField)('__element_idx),
```

The post-aggregation operation needs to have `sa` in scope, so I need a way to expose a per-element context. The simplest thing was to bind the element index in the post-aggregation scope, and this is the emitted code to do that.